server/modules/nodeDef/service/nodeDefService.js

```diff
@@ -1,6 +1,134 @@
+import * as Survey from '@core/survey/survey'
+import * as NodeDef from '@core/survey/nodeDef'
+import * as SurveyValidator from '@core/survey/surveyValidator'
+import * as Validation from '@core/validation/validation'
+
+import { db } from '@server/db/db'
+import * as SurveyManager from '@server/modules/survey/manager/surveyManager'
 import * as NodeDefManager from '../manager/nodeDefManager'
 
-export const { insertNodeDef, updateNodeDefProps, markNodeDefDeleted } = NodeDefManager
+const fetchSurvey = async ({ surveyId, cycle }, t) => {
+  const surveyDb = await SurveyManager.fetchSurveyAndNodeDefsBySurveyId(
+    {
+      surveyId,
+      cycle,
+      draft: true,
+      advanced: true,
+      validate: false,
+    },
+    t
+  )
+  // build dependency graph if empty
+  return Survey.hasDependencyGraph(surveyDb) ? surveyDb : Survey.buildAndAssocDependencyGraph(surveyDb)
+}
+
+const afterNodeDefUpdate = async ({ survey, nodeDef, nodeDefsDependent = [], nodeDefsUpdated }, t) => {
```

Comment:
do the t should be "client = db"

Reply:
it is an internal function, I didn't want to change too much the code, but yes, you are right